### PR TITLE
Load ReactiveJwtAuthenticationConverter bean in OAuth2 Resource Server config

### PR DESCRIPTION
**Pull request by ThomasVitale**
When a bean of type ReactiveJwtAuthenticationConverter is defined,
the OAuth2 Resource Server configuration will use it automatically
when no other converter is defined through the DSL.

Closes gh-9698

### CODE REVIEW

**Comment by jzheaux** [comment]
Thanks for the PR, @ThomasVitale! I've scheduled this for the 5.6 release.

**Comment by ThomasVitale** [comment]
@jzheaux Nice! Thanks for checking the PR.

File: config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java [hunk 1]
```diff
@@ -478,6 +479,59 @@ public void getJwtDecoderWhenNoBeansAndNoDslWiredThenWiringException() {
...
+		assertThat(jwt.getJwtAuthenticationConverter()).isNotEqualTo(beanWiredJwtAuthenticationConverter);
+	}
+
+	@Test
+	public void getJwtAuthenticationConverterWhenNoBeansAndNoDslWiredThenDefaultConverter() {
+		GenericWebApplicationContext context = autowireWebServerGenericWebApplicationContext();
+		ServerHttpSecurity http = new ServerHttpSecurity();
+		http.setApplicationContext(context);
+		ServerHttpSecurity.OAuth2ResourceServerSpec.JwtSpec jwt = http.oauth2ResourceServer().jwt();
+		assertThat(jwt.getJwtAuthenticationConverter()).isNotNull();
```
Comment (jzheaux) [inline config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java:532]: Should this instead test that it's an instanceof `ReactiveJwtAuthenticationConverter`?

File: config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java [hunk 2]
```diff
@@ -478,6 +479,59 @@ public void getJwtDecoderWhenNoBeansAndNoDslWiredThenWiringException() {
...
+				() -> beanWiredJwtAuthenticationConverter);
+		context.registerBean("secondJwtAuthenticationConverter", ReactiveJwtAuthenticationConverter.class,
+				() -> beanWiredJwtAuthenticationConverter);
+		ServerHttpSecurity.OAuth2ResourceServerSpec.JwtSpec jwt = http.oauth2ResourceServer().jwt();
+		jwt.jwtAuthenticationConverter(dslWiredJwtAuthenticationConverter);
+		assertThat(jwt.getJwtAuthenticationConverter()).isEqualTo(dslWiredJwtAuthenticationConverter);
+	}
+
+	@Test
+	public void getJwtAuthenticationConverterWhenTwoBeansWiredThenDefaultTakesPrecedence() {
```
Comment (jzheaux) [inline config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java:512]: I'd expect this test to align with `getJwtDecoderWhenTwoBeansWiredThenThrowsWiringException` and `OAuth2ResourceServerConfigurerTests#getJwtAuthenticationConverterWhenDuplicateConverterBeansThenThrowsException`. Is there a reason that this diverges from those?

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @ThomasVitale! Please see my inline feedback.

**Comment by jzheaux** [comment]
Hi, @ThomasVitale! Are you able to apply the requested changes?

**Comment by ThomasVitale** [comment]
@jzheaux sorry for the late reply. I can deliver the requested changes by the end of this week. Would that be ok?

**Comment by jzheaux** [comment]
Thanks for checking in, @ThomasVitale. Yes, that works great.

File: config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java (hunk 2 above)
Comment (ThomasVitale) [reply config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java:512]: That's an interesting question. The reason is that `ReactiveJwtAuthenticationConverter` is resolved through the `getBeanOrNull()` method defined in `ServerHttpSecurity`. If more than one bean is defined, the logic will use the default one and not throw an error. I based the implementation on how most of the beans for OAuth2 and OIDC are configured in that class (e.g. `OidcUserService` or `ReactiveClientRegistrationRepository`). You made me notice that, in the imperative version, we throw an error if there's more than one bean defined. I guess we want the same logic for the reactive version as well. Thoughts?

Comment (jzheaux) [reply config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java:512]: Yes, that was my thinking. 

If an application declares more than one `ReactiveJwtAuthenticationConverter` bean, the application should clearly indicate which one should be used. An error is a simple way to do that.

File: config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java (hunk 1 above)
Comment (ThomasVitale) [reply config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java]: Good point. Changed to check for the instance type.

File: config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java (hunk 2 above)
Comment (ThomasVitale) [reply config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java]: Perfect! I changed it to the agreed behaviour.

**Review by jzheaux — APPROVED** [review]

**Pull request merged**